cartographer/mapping/trajectory_node.h

```diff
@@ -56,6 +57,8 @@ struct TrajectoryNode {
   transform::Rigid3d pose;
 };
 
+proto::TrajectoryNode ToProto(const TrajectoryNode::Data& constant_data);
```

Comment:
I don't see FromProto. Where are you planning to deserialize the data? Add FromProto and a round-trip unit test?